pkg/generator/generator_test.go

```diff
@@ -446,8 +446,9 @@ func main() {
 		logrus.Fatalf("failed to get watch namespace: %v", err)
 	}
 	resyncPeriod := 5
+	numWorkers := 1
 	logrus.Infof("Watching %s, %s, %s, %d", resource, kind, namespace, resyncPeriod)
-	sdk.Watch(resource, kind, namespace, resyncPeriod)
+	sdk.Watch(resource, kind, namespace, resyncPeriod, numWorkers)
```

Comment:
I would also like to understand how this might change with regards to controller runtime. Especially, before making a backward incompatible change and then maybe needing to do another after the controller runtime refactor.

Reply:
use  this https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md

Reply:
@shawn-hurley 
`I would also like to understand how this might change with regards to controller runtime`
Currently, we are not sure how `controller runtime` affects the overall API change. we will deal with it when it comes.

Reply:
@shawn-hurley The controller-runtime supports configuring the number of workers out of the box via the controller options:
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controller.go#L32-L34

So this feature should be supported eventually once we use the controller-runtime.
But in the meantime if we can still support this without breaking backwards compatibility then it's worth adding right now since this has been requested for quite some time.